### PR TITLE
Create data_visualization_seaborn

### PR DESCRIPTION
**Pull request by pm0kjp**
New repo, let's merge this in!

### CODE REVIEW

**Comment by rosemm** [comment]
This one is still under construction and shouldn't be merged yet

**Comment by rosemm** [comment]
Okie dokie, this one is ready for review now!

File: data_visualization_in_seaborn/data_visualization_seaborn.md [hunk 1]
```diff
@@ -0,0 +1,684 @@
...
+
+For more background on `regplot` vs. `lmplot`, see [the seaborn regression tutorial](https://seaborn.pydata.org/tutorial/regression.html#controlling-the-size-and-shape-of-the-plot).
+</div>
+
+```python
+sns.lmplot(data = covid_data,
+            x="val_age", y="val_height_cm")
+```
+
+![Scatterplot with age on the x-axis and height on the y-axis, and a straight trend line running through the points, with a lighter shaded band around it.](media/seaborn_trend_1.png)
```
Comment (franzenr) [inline data_visualization_in_seaborn/data_visualization_seaborn.md:538]: more context/description of what's actually communicated via the image(e.g., mention the trend line fit to underlying data which you get into later)

File: data_visualization_in_seaborn/data_visualization_seaborn.md [hunk 2]
```diff
@@ -0,0 +1,684 @@
...
+
+A couple things about this plot are not ideal:
+
+1. Overplotting. The density of the scatter plot makes it hard to see the trend line as it runs through the bulk of the data.
+2. The linear trend doesn't appear to be a very good description of the relationship between age and height. It systematically over-estimates height at the very young ages, then under-estimates through early adulthood and over-estimates again late in life.
+
+We can address the first issue by adjusting alpha for the scatterplot. Alpha ranges between 0 (totally transparent) and 1 (totally opaque).
```
Comment (franzenr) [inline data_visualization_in_seaborn/data_visualization_seaborn.md:545]:
```suggestion
We can address the first issue by adjusting alpha for the scatterplot to make the markers more transparent. Alpha ranges between 0 (totally transparent) and 1 (totally opaque).
```

File: data_visualization_in_seaborn/data_visualization_seaborn.md [hunk 3]
```diff
@@ -0,0 +1,684 @@
...
+
+<div class = "important">
+Tip: When your data overlap too much on a plot, use alpha to make them more transparent.
+</div>
```
Comment (franzenr) [inline data_visualization_in_seaborn/data_visualization_seaborn.md:549]:
```suggestion
```

File: data_visualization_in_seaborn/data_visualization_seaborn.md [hunk 4]
```diff
@@ -0,0 +1,684 @@
...
+
+We face a problem here, though: Since we're using a single function to draw both the scatter plot and the trend line, how do we adjust alpha for the scatter plot without affecting the line?
```
Comment (franzenr) [inline data_visualization_in_seaborn/data_visualization_seaborn.md:551]: they haven't done anything in terms of adjusting alpha anywhere yet, have they? I'm not sure they would have a framework yet for why it would cause difficulty to adjust the alpha for just the plot. For all they know, perhaps there's an argument for line alpha vs scatterplot alpha in lmplot.

File: data_visualization_in_seaborn/data_visualization_seaborn.md [hunk 5]
```diff
@@ -0,0 +1,684 @@
...
+
+1. Overplotting. The density of the scatter plot makes it hard to see the trend line as it runs through the bulk of the data.
+2. The linear trend doesn't appear to be a very good description of the relationship between age and height. It systematically over-estimates height at the very young ages, then under-estimates through early adulthood and over-estimates again late in life.
+
+We can address the first issue by adjusting alpha for the scatterplot. Alpha ranges between 0 (totally transparent) and 1 (totally opaque).
+
+<div class = "important">
+Tip: When your data overlap too much on a plot, use alpha to make them more transparent.
+</div>
+
```
Comment (franzenr) [inline data_visualization_in_seaborn/data_visualization_seaborn.md:550]: consider a page break here? definitely not necessary, but this page seems longer than many of the others, and since there's a lot of new content on this page, perhaps having that break would be useful!

Also, since addressing point 2 (trend line fit) moves on to another page, maybe it would make sense to have the content addressing point 1 separated out as well.

File: data_visualization_in_seaborn/data_visualization_seaborn.md [hunk 6]
```diff
@@ -0,0 +1,684 @@
...
+We'll try adding a quadratic term here, to see if it looks like a second order polynomial linear regression is a better description of the data.
+
+```python
+sns.lmplot(data = covid_data,
+            x="val_age", y="val_height_cm",
+           scatter_kws={"alpha": .1},
+           order = 2)
+```
+
+![The scatter plot with age on the x-axis and height on the y-axis, but now the trend line drawn is a upside-down U shape.](media/seaborn_trend_3.png)
```
Comment (franzenr) [inline data_visualization_in_seaborn/data_visualization_seaborn.md:586]:
```suggestion
![The scatter plot with age on the x-axis and height on the y-axis, but now the trend line drawn is a upside-down U shape, suggesting that the general trend is for height to increase with age until it reaches some maximum, at which point height begins decreasing.](media/seaborn_trend_3.png)
```

File: data_visualization_in_seaborn/data_visualization_seaborn.md [hunk 7]
```diff
@@ -0,0 +1,684 @@
...
+
+This appears to be a slight improvement on the linear regression model, especially at the youngest ages, but it is still systematically over-estimating height for young children and, because the quadratic trend follows a smooth arc shape, it predicts a drop in height at the older ages which is neither supported by the data nor sensible from a theoretical standpoint. A model that suggests height peaks around 40 years of age and then drops off dramatically does not make practical sense.
```
Comment (franzenr) [inline data_visualization_in_seaborn/data_visualization_seaborn.md:588]: this feels a bit unfinished in that we don't seem to get to what _would_ be an appropriate line for this data (which, of course, is likely outside the scope of the module, but it feels like it's implied to be covered by this paragraph)

File: data_visualization_in_seaborn/data_visualization_seaborn.md [hunk 8]
```diff
@@ -0,0 +1,684 @@
...
+
+Modify the code from the final example, the [lowess curve trend line](#lowess-curve-trend-lines), to get separate trend lines for each smoking status (`is_smoker`), but all shown together on single plot. (Hint: Use color to distinguish the groups.)
+
+```r  -Solution
+sns.lmplot(data = covid_data,
+            x="val_age", y="val_height_cm",
+           scatter_kws={"alpha": .1},
+           lowess=True,
+           hue = 'is_smoker')
+```
```
Comment (franzenr) [inline data_visualization_in_seaborn/data_visualization_seaborn.md:648]: not sure if it's clear in this section that they should be going back to the ipynb file. Perhaps make that explicit with a quick instruction?

File: data_visualization_in_seaborn/data_visualization_seaborn.md [hunk 9]
```diff
@@ -0,0 +1,684 @@
...
+****
+
+Modify the code from the final example, the [lowess curve trend line](#lowess-curve-trend-lines), to separate out respondents by smoking status (`is_smoker`) with a separate facet for each.
+
+```r  -Solution
+sns.lmplot(data = covid_data,
+            x="val_age", y="val_height_cm",
+           scatter_kws={"alpha": .1},
+           lowess=True,
+           col = 'is_smoker')
```
Comment (franzenr) [inline data_visualization_in_seaborn/data_visualization_seaborn.md:637]: are there any hangups you can anticipate for learners in terms of them figuring out this solution/understanding _why_ this is the solution? If so, perhaps a bit of explanation as to why e.g., this uses `col` and not `row` could be helpful

File: data_visualization_in_seaborn/data_visualization_seaborn.md [hunk 10]
```diff
@@ -0,0 +1,684 @@
...
+```
+
+Modify the code from the final example, the [lowess curve trend line](#lowess-curve-trend-lines), to get separate trend lines for each smoking status (`is_smoker`), but all shown together on single plot. (Hint: Use color to distinguish the groups.)
+
+```r  -Solution
+sns.lmplot(data = covid_data,
+            x="val_age", y="val_height_cm",
+           scatter_kws={"alpha": .1},
+           lowess=True,
+           hue = 'is_smoker')
```
Comment (franzenr) [inline data_visualization_in_seaborn/data_visualization_seaborn.md:647]: same thought as above in terms of adding a bit more explanation for why this is the correct code

File: data_visualization_in_seaborn/data_visualization_seaborn.md [hunk 11]
```diff
@@ -0,0 +1,684 @@
...
+```
+
+## Additional Resources
+
+The creator of `seaborn`, [Michael Waskom](https://mwaskom.github.io/) maintains an excellent website with examples, explanations, tutorials, and more: [https://seaborn.pydata.org/](https://seaborn.pydata.org/)
```
Comment (franzenr) [inline data_visualization_in_seaborn/data_visualization_seaborn.md:652]:
```suggestion
The creator of `seaborn` [Michael Waskom](https://mwaskom.github.io/) maintains an [excellent website](https://seaborn.pydata.org/) with examples, explanations, tutorials, and more.
```

File: data_visualization_in_seaborn/data_visualization_seaborn.md [hunk 12]
```diff
@@ -0,0 +1,684 @@
...
+[[col, row]]
+<script>
+  let input = "@input".trim();
+  /col|row/i.test(input);
+</script>
+****
+You can add facets to a plot by including the arguments for `col`, `row`, or both.
+****
+
+## Trend Lines
```
Comment (franzenr) [inline data_visualization_in_seaborn/data_visualization_seaborn.md:517]: In general, this final section feels a bit more stats-y / viz theory heavy than the other sections, which felt like they were more about the mechanics of changing various things using code. Relatedly, it feels like it assumes a higher level of background stats than other sections by including things like lowess, which many learners may not be particularly familiar with before reading this. 

This isn't necessarily a good or a bad thing, just something I noticed for you to think about either way.

File: data_visualization_in_seaborn/data_visualization_seaborn.md [hunk 13]
```diff
@@ -0,0 +1,684 @@
...
+
+<div class = "care">
+There is a much steeper learning curve for `matplotlib` compared to `seaborn`, however, and you may find you're happy with most `seaborn` plots without needing additional control --- don't bother with it if it feels too overwhelming. When you're ready to read more about `matplotlib`, start with their [introductory tutorials](https://matplotlib.org/stable/tutorials/index).
+</div>
+
+In this case, we can add an argument `scatter_kws` that passes additional "keyword arguments" to the `matplotlib` function drawing the scatter plot via a [python dictionary](https://www.w3schools.com/python/python_dictionaries.asp). There are [many possible keyword arguments](https://matplotlib.org/3.5.1/api/_as_gen/matplotlib.lines.Line2D.html#matplotlib.lines.Line2D) you can use, but we'll just set `alpha` here.
```
Comment (franzenr) [inline data_visualization_in_seaborn/data_visualization_seaborn.md:561]: is it worthwile to explain what the scatter_kws argument is and why we pass it things in a dictionary, or will that get way too into the weeds very quickly?

File: data_visualization_in_seaborn/data_visualization_seaborn.md [hunk 14]
```diff
@@ -0,0 +1,684 @@
...
+- [Exploring data using pandas and seaborn](https://colab.research.google.com/github/arcus/education-materials/blob/master/explore-pandas-seaborn/explore-pandas-seaborn-full.ipynb)
+- [Intro to machine learning, Part 1: Classification with Scikit-learn](https://colab.research.google.com/github/arcus/education-materials/blob/master/intro-to-ml/ml_workbook_00_intro.ipynb) (note, this pairs with an article on the Arcus Education team website: [What type of machine learning should I use?](https://education.arcus.chop.edu/types-of-ml/))
+- [Intro to machine learning, Part 2: Validation, tuning, and model selection](https://colab.research.google.com/github/arcus/education-materials/blob/master/intro-to-ml/ml_workbook_01_validation_tuning.ipynb)
+- [Intermediate machine learning](https://github.com/arcus/education-materials/tree/master/ml-intermediate) (scroll down to the README description for links to several interactive notebooks)
+- [Network analysis and visualization with NetworkX](https://colab.research.google.com/github/arcus/education-materials/blob/master/network-viz/introduction.ipynb)
+
```
Comment (franzenr) [inline data_visualization_in_seaborn/data_visualization_seaborn.md:665]: are these all publicly available and intended to be shared with the broad public (so not just off-network CHOP folks but literally anyone in the world whenever?)

**Review by franzenr — CHANGES_REQUESTED** [review]
hey @rosemm 

Just finished working through this first round of reviews. let me know what you think.

File: data_visualization_in_seaborn/data_visualization_seaborn.md [hunk 15]
```diff
@@ -0,0 +1,684 @@
...
+```python
+import pandas as pd
+import numpy as np
+import matplotlib.pyplot as plt
+import seaborn as sns
+
+```
+
+<div class = "learnmore">
```
[the remark this answers is not in the document]
Comment (rosemm) [reply data_visualization_in_seaborn/data_visualization_seaborn.md:103]: Yeah, I think that's a good idea. It will be a change in the css file, though, not this module. So I'm resolving it here for the purposes of this review.

File: data_visualization_in_seaborn/data_visualization_seaborn.md [hunk 16]
```diff
@@ -0,0 +1,684 @@
...
+email:    hartmanr1@chop.edu
+version:  0.0.1
+language: en
+narrator: UK English Female
+title: Data visualizations in seaborn
+
+comment:  This module includes code and explanations for several popular data visualizations, using python's seaborn library. It also includes examples of how to modify seaborn plots to customize them for different uses.  
+
+long_description: You can use the seaborn module in python to make many different kinds of data visualizations (also called plots, or charts), including scatterplots, histograms, line plots, and trend lines. This module provides an example of each of these kinds of plots, including python code to make them using the seaborn module. It may be hard to follow if you are brand new to python, but it is appropriate for beginners with at least a small amount of python experience.
+
```
[the remark this answers is not in the document]
Comment (rosemm) [reply data_visualization_in_seaborn/data_visualization_seaborn.md:14]: Done!

File: data_visualization_in_seaborn/data_visualization_seaborn.md [hunk 17]
```diff
@@ -0,0 +1,684 @@
...
+## Lesson Preparation
+
+
+For the hands-on activities in this module, we will be using an online environment containing a Jupyter notebook. You don't need to install anything or set up an account, but you need a modern web browser like Chrome and a moderately good wifi connection.
+
+Because it can take a few minutes for the environment to be created, we suggest you click the link below to start up the activity. We recommend using right-click to open it in a new tab or window, and then returning here to continue learning while the environment finishes loading. Here is the link:
+
+<a href = "https://mybinder.org/v2/gh/arcus/education_modules/data_viz_seaborn?labpath=data_visualization_in_seaborn%2Fnotebooks%2Fdata_visualization_seaborn.ipynb" target = "_blank"> <img src = "https://mybinder.org/badge_logo.svg"></a> **← Click the "launch binder" button!**
+
+If you have python already installed on your computer and you prefer to work through code examples there, you can <a href="https://raw.githubusercontent.com/arcus/education_modules/main/data_visualization_in_seaborn/notebooks/data_visualization_seaborn.ipynb" download>download the code for this module to run offline</a>.
```
[the remark this answers is not in the document]
Comment (rosemm) [reply data_visualization_in_seaborn/data_visualization_seaborn.md:67]: Done!

File: data_visualization_in_seaborn/data_visualization_seaborn.md [hunk 18]
```diff
@@ -0,0 +1,684 @@
...
+covid_data = pd.read_csv("https://raw.githubusercontent.com/kendavidn/yaounde_serocovpop_shared/v1.0.0/data/yaounde_covid_seroprev_dataset.csv")
+
+```
+
+These data are from a COVID-19 serological survey conducted in Yaounde, Cameroon (Nwosu, K., Fokam, J., Wanda, F. et al., 2021[^1](Kene David Nwosu, Joseph Fokam, Franck Wanda, Lucien Mama, Erol Orel, Nicolas Ray, Jeanine Meke, Armel Tassegning, Desire Takou, Eric Mimbe, Beat Stoll, Josselin Guillebert, Eric Comte, Olivia Keiser, & Laura Ciaffi. (2021). kendavidn/yaounde_serocovpop_shared: Initial release (v1.0.0). Zenodo. https://doi.org/10.5281/zenodo.5218965)). The authors have made all of the code and data publicly available under a [creative commons 4.0 license](https://creativecommons.org/licenses/by/4.0/legalcode) to facilitate re-use.
+
+
+<div class="learnmore">
+To learn more about the study, see the [zenodo page for this dataset](https://zenodo.org/record/5218965#.YeBq2RPMITW). You can read the published article online: [SARS-CoV-2 antibody seroprevalence and associated risk factors in an urban district in Cameroon](https://www.nature.com/articles/s41467-021-25946-0).
+</div>
```
[the remark this answers is not in the document]
Comment (rosemm) [reply data_visualization_in_seaborn/data_visualization_seaborn.md:123]: The second zenodo link is actually to information about the data itself (code book, etc.) rather than about the study, which is the first zenodo link. I think it makes sense to keep both. 
Good catch on the escapes --- thanks!
Thanks for the note about the highlight boxes --- I'm looking through again and will make sure I'm only putting in ones I definitely think should be there.

File: data_visualization_in_seaborn/data_visualization_seaborn.md [hunk 19]
```diff
@@ -0,0 +1,684 @@
...
+Tip: It's generally much easier to make any necessary changes to the dataframe, such as modifying variables, before sending it to the plotting command.
+</div>
+
+### Distinguish groups more clearly with color and shape
+
+We'll improve this plot by using shape and color together to mark the Classification groups.
+
+<div class = "important">
+Tip: Don't use color alone to convey important information in your plots because if your end users are unable to distinguish the colors, the plot loses its value. Instead, double-up color information with another element, such as marker shape, to make the different groups easier to distinguish.
+</div>
```
[the remark this answers is not in the document]
Comment (rosemm) [reply data_visualization_in_seaborn/data_visualization_seaborn.md:206]: I'm leaning toward keeping this as a highlight box, but thanks for the suggestion. I have clarified the code explanation, so hopefully that lands better now.

File: data_visualization_in_seaborn/data_visualization_seaborn.md [hunk 20]
```diff
@@ -0,0 +1,684 @@
...
+
+### Controlling figure aesthetics
+
+The seaborn library handles the appearance of plots with two kinds of functions: style, and context.
+
+- `style` refers to things like background color, gridlines, and axis ticks
+- `context` refers to the visual scale of the plot, such as the size of the font used for axis labels
+
+Seaborn controls style and context separately so that you can get a plot that has all the style elements you want, and then you can present that same plot scaled appropriately in different contexts --- for example, if you want to include it in a slide deck presentation, you probably want the fonts much larger and the lines heavier than you would if you wanted to include it as a figure in a paper. In this case, you would leave all of the style settings the same, but change context to "talk" to scale it for the slide deck and "paper" to scale it for use in the paper.
```
[the remark this answers is not in the document]
Comment (rosemm) [reply data_visualization_in_seaborn/data_visualization_seaborn.md:249]: Done

File: data_visualization_in_seaborn/data_visualization_seaborn.md [hunk 21]
```diff
@@ -0,0 +1,684 @@
...
+Because histograms show just one variable, the only aesthetic they require is x. The y-axis of the plot will just show the counts of observations in each bin on the x-axis. (Note that it is possible to provide `sns.displot` with y instead of x, in which case it will generate a sideways histogram.)
+
+```python
+sns.displot(covid_data, x="val_age")
+```
+
+![Histogram showing the distribution of age values.](media/seaborn_hist_1.png)
+
+
+### Change the number of bins
```
[the remark this answers is not in the document]
Comment (rosemm) [reply data_visualization_in_seaborn/data_visualization_seaborn.md:329]: This is mentioned later, and a link provided for more details for anyone who wants to dig into it.

File: data_visualization_in_seaborn/data_visualization_seaborn.md [hunk 22]
```diff
@@ -0,0 +1,684 @@
...
+
+The appearance of a histogram can change a lot depending on the number of bins you use along the x-axis. It's a good idea to try a few different sets of bins to see what works well for communicating this distribution.
+
+```python
+sns.displot(covid_data, x="val_age", binwidth=1)
+
+```
+
+![Histogram showing the distribution of ages with many narrow bins, each representing a single year.](media/seaborn_hist_2.png)
```
[the remark this answers is not in the document]
Comment (rosemm) [reply data_visualization_in_seaborn/data_visualization_seaborn.md:338]: Added to main text rather than alt text, since I thought it was a suggestion that everybody would benefit from, including folks with visual access to the plots :)

File: data_visualization_in_seaborn/data_visualization_seaborn.md [hunk 23]
```diff
@@ -0,0 +1,684 @@
...
+As with scatterplots, we can add information about an additional variable by using color. Let's put in the variable `is_smoker` for hue so we can see how the distribution of ages differs by smoking status.
+
+Note that seaborn is continuing to use the color palette we set when we were making scatter plots (see [custom colors](#custom-colors)). If you want to change the color palette, you can do so at any time.
+
+```python
+sns.displot(covid_data, x="val_age", hue="is_smoker")
+
+```
+
+![Histogram showing the distribution of ages, with non-smokers in blue, smokers in orange, and ex-smokers in green, overlaid.](media/seaborn_hist_4.png)
```
[the remark this answers is not in the document]
Comment (rosemm) [reply data_visualization_in_seaborn/data_visualization_seaborn.md:357]: It's not in the alt text, but I feel like it's covered in the main text --- let me know if you disagree.

File: data_visualization_in_seaborn/data_visualization_seaborn.md [hunk 24]
```diff
@@ -0,0 +1,684 @@
...
+That's often the most useful way to plot multiple histograms, but in this case, it makes it hard to see the smoker and ex-smoker distributions clearly because they fall completely within the non-smoker distribution.
+
+You can control how `seaborn` plots the distributions with the `multiple` argument. To show stacked groups instead, use `multiple = "stack"`. This will put the three distributions one on top of the other, so the height of all three colors together in each bin corresponds to the total count across groups; the silhouette of this histogram will be identical to the original histogram without coloring by group.
+
+
+```python
+sns.displot(covid_data, x="val_age", hue="is_smoker", multiple="stack")
+```
+
+![Histogram showing the distribution of ages, with non-smokers in blue, smokers in orange, and ex-smokers in green, stacked.](media/seaborn_hist_5.png)
```
[the remark this answers is not in the document]
Comment (rosemm) [reply data_visualization_in_seaborn/data_visualization_seaborn.md:370]: Again, I feel like this is covered in the main text --- let me know if you think I'm missing something.

File: data_visualization_in_seaborn/data_visualization_seaborn.md [hunk 25]
```diff
@@ -0,0 +1,684 @@
...
+Not necessarily! The `seaborn` defaults are sensible, so that's often a good starting place, but it can be dangerous to accept the default bins without examining other options.  Changing the binwidth on a histogram can dramatically change the patterns highlighted in the data, so it's always a good idea to try multiple binwidths when you're making a histogram.
+
+You can read more about this issue in the [seaborn distributions tutorial](https://seaborn.pydata.org/tutorial/distributions.html#choosing-the-bin-size).
+****
+
+## Line Plots
+
+Line plots are especially useful when you want to show data points that are connected in a meaningful way. The most common application is repeated measures over time (also called time series data), such as when patients are measured on a given variable (plotted on the y-axis) at several times (plotted along the x-axis), and each line would represent one patient, or a summary across a group of patients.
+
+<div class = "important">
```
[the remark this answers is not in the document]
Comment (rosemm) [reply data_visualization_in_seaborn/data_visualization_seaborn.md:436]: Good idea

File: data_visualization_in_seaborn/data_visualization_seaborn.md [hunk 26]
```diff
@@ -0,0 +1,684 @@
...
+```Python
+fmri = sns.load_dataset("fmri")
+```
+
+### Basic line plot
+
+```python
+sns.relplot(x="timepoint", y="signal", kind="line", data=fmri)
+```
+![Basic line plot showing the strength of the FMRI signal over time, with timepoint on the x-axis and signal on the y-axis.](media/seaborn_line_1.png)
```
[the remark this answers is not in the document]
Comment (rosemm) [reply data_visualization_in_seaborn/data_visualization_seaborn.md:455]: Done

File: data_visualization_in_seaborn/data_visualization_seaborn.md [hunk 27]
```diff
@@ -0,0 +1,684 @@
...
+
+Note that by default `seaborn` aggregates across multiple observations at each time point, with a line at the mean and a 95% confidence interval shaded around it.
+
+In this case, we have several different subjects, and from each we have readings from two brain regions and in response to two events, so there are many observations at each timepoint. If there were only a single observation for each timepoint (more typical of standard time series data), then `relplot` would show a line connecting those observations and no confidence interval.
```
[the remark this answers is not in the document]
Comment (rosemm) [reply data_visualization_in_seaborn/data_visualization_seaborn.md:459]: I'm actually going to keep the optional "and" in there because I want to emphasize that these two things correspond to two different variables in the data

File: data_visualization_in_seaborn/data_visualization_seaborn.md (hunk 12 above)
[the remark this answers is not in the document]
Comment (rosemm) [reply data_visualization_in_seaborn/data_visualization_seaborn.md:528]: Yeah, I see what you mean, thanks! For now, I'm going to keep the ordering as-is (I prefer having trend lines at the end for the reasons you state, and I don't think it will be too taxing for them to switch between datasets).

Comment (rosemm) [reply data_visualization_in_seaborn/data_visualization_seaborn.md:528]: Thanks for the heads up --- I'll read through it again with an eye toward potential pitfalls, and things that should maybe be added to the prereqs.

File: data_visualization_in_seaborn/data_visualization_seaborn.md (hunk 3 above)
Comment (rosemm) [reply data_visualization_in_seaborn/data_visualization_seaborn.md:560]: Done

File: data_visualization_in_seaborn/data_visualization_seaborn.md (hunk 4 above)
Comment (rosemm) [reply data_visualization_in_seaborn/data_visualization_seaborn.md:562]: I clarified this --- lemme know if you still think it's confusing.

File: data_visualization_in_seaborn/data_visualization_seaborn.md (hunk 13 above)
Comment (rosemm) [reply data_visualization_in_seaborn/data_visualization_seaborn.md:572]: Haha yeah, I was thinking that was perhaps too weedy. The links provided will guide folks who want to dig further into it, but I think for the actual module content it makes sense to just focus on how to implement this particular change (adjusting alpha) rather than opening the pandora's box of matplotlib arguments.

File: data_visualization_in_seaborn/data_visualization_seaborn.md (hunk 9 above)
Comment (rosemm) [reply data_visualization_in_seaborn/data_visualization_seaborn.md:648]: Done

File: data_visualization_in_seaborn/data_visualization_seaborn.md (hunk 8 above)
Comment (rosemm) [reply data_visualization_in_seaborn/data_visualization_seaborn.md:659]: Good call, added a little highlight box for additional guidance here.

File: data_visualization_in_seaborn/data_visualization_seaborn.md (hunk 10 above)
Comment (rosemm) [reply data_visualization_in_seaborn/data_visualization_seaborn.md:658]: Done

File: data_visualization_in_seaborn/data_visualization_seaborn.md (hunk 14 above)
Comment (rosemm) [reply data_visualization_in_seaborn/data_visualization_seaborn.md:676]: I'm not sure, so I'm just deleting them for now. We can revisit that later, perhaps.

File: data_visualization_in_seaborn/data_visualization_seaborn.md [hunk 28]
```diff
@@ -0,0 +1,684 @@
...
+
+<div class="learnmore">
+To learn more about the study, see the [zenodo page for this dataset](https://zenodo.org/record/5218965#.YeBq2RPMITW). You can read the published article online: [SARS-CoV-2 antibody seroprevalence and associated risk factors in an urban district in Cameroon](https://www.nature.com/articles/s41467-021-25946-0).
+</div>
+
+<div class="important">
+Run the above code yourself in binder (see [lesson preparation](#lesson-preparation) for links to start the binder instance) or on your own computer.
+
+In the data\_visualization\_seaborn.ipynb file, the code in the **Setup** section includes these import commands and the command to read the csv file for the data. Before you will be able to generate the plots in the rest of the module, you should run those lines of code.
+</div>
```
[the remark this answers is not in the document]
Comment (rosemm) [reply data_visualization_in_seaborn/data_visualization_seaborn.md]: Removed the highlight. 
About repeating info, I think it's valuable right here since this is the point at which they need to actually start using the binder instance they hopefully set up before. Initial instructions for binder say to just click it then come back here and let it work, so I think it's worthwhile to be super clear that **this** is the point at which they should be actually working in binder in addition to reading through the module. The instructions are repeated on purpose to be extra clear and make sure we don't lose folks at this crucial juncture.

**Comment by rosemm** [comment]
Okay, @franzenr it's ready for you again! Thanks! 😄

**Comment by rosemm** [comment]
Okay @drelliche I think it's all set now (although I'm still seeing the old plots with the white background when I test it, I assume that's a caching issue since there's zero linking happening now). So maybe it needs a day to rest or so, but hopefully it will be a very speedy check after that. 🎉

**Comment by pm0kjp** [comment]
@rosemm, @drelliche: Ironically I can't actually approve this one because the PR originated with me!  But I did do a spot check and it renders good for me, not sure what the "white background" @rosemm mentions refers to but at least the first 18 pages were great so I'm making an executive decision and yanking this one through without approval (although I do approve!)
